src/services/simulation.ts

```diff
@@ -74,4 +84,79 @@ export class SimulationService extends Service {
 
     return response;
   }
+
+  async simulateZapIn(
+    from: Address,
+    token: Address,
+    amount: Integer,
+    vault: Address,
+    gasPrice: Integer,
+    slippagePercentage: number,
+    block: number
+  ): Promise<TransactionOutcome> {
+    const url = "https://api.zapper.fi/v1/zap-in/yearn/transaction";
+    const params = new URLSearchParams({
+      ownerAddress: from,
+      sellTokenAddress: token,
+      sellAmount: amount,
+      poolAddress: vault,
+      gasPrice: gasPrice,
+      slippagePercentage: slippagePercentage.toString(),
+      api_key: this.ctx.zapper
+    });
+    const zapInParams: ZapInOutput = await fetch(`${url}?${params}`).then(res => res.json());
+
+    const body = {
+      network_id: "1",
+      block_number: block,
+      transaction_index: 0,
+      from: from,
+      input: zapInParams.data,
+      to: zapInParams.to,
+      gas: zapInParams.gas,
+      simulation_type: "quick",
+      gas_price: zapInParams.gasPrice,
+      value: zapInParams.value,
+      save: true
+    };
+
+    interface SimulationCallTrace {
+      output: string;
+    }
+
+    interface SimulationTransactionInfo {
+      call_trace: SimulationCallTrace;
+    }
+
+    interface SimulationTransaction {
+      transaction_info: SimulationTransactionInfo;
+    }
+
+    interface SimulationResponse {
+      transaction: SimulationTransaction;
+    }
+
+    const response: SimulationResponse = await fetch(`${baseUrl}/simulate`, {
```

Comment:
Kinda confused how to get the same value as it's getting from the UI, because there's a lot of nesting to work around atm.

<img width="423" alt="Screenshot 2021-06-29 at 23 21 43" src="https://user-images.githubusercontent.com/19509999/123874950-ca435500-d930-11eb-9231-b1c183733a06.png">